react/src/pages/datasets/DatasetTable.tsx

```diff
@@ -87,7 +89,10 @@ export default function DatasetTable() {
                 <DatasetInfoDialog
                     dataset_id={infoDataset.dataset_id}
                     open={showInfo}
-                    onClose={() => setShowInfo(false)}
+                    onClose={() => {
+                        setShowInfo(false);
+                        reRender(n => n + 1);
```

Comment:
The table should only be re-rendered if the user changed something in the dialog. You should also look into re-using material-table onRowUpdate for updating/redrawing the table, as material-table probably has a more optimized means of redrawing new rows.